### PR TITLE
Reuse exception in conn finalizer and set __FILE__ and __LINE__

### LINKED ISSUE #120: [Neo] ConnectionError notification exception is missing file and line information.
When using the NotificationFormatter output for the following code, i noticed that the log error showed `@ :0`

```
    private void connNotifier ( DhtClient.Neo.DhtConnNotification info )
    {
        auto message = formatNotification(info, this.notification_buffer);
        with ( info.Active ) switch ( info.active )
        {
            case connected:
            case hash_range_queried:
                this.log.info(message);
                break;
            default:
                this.log.error(message);
        }
    }
```

`<connection_error>: Exception 'ConnectProtocol: finalized' @ :0 occurred in the client while handling the request on node [127, 0, 0, 1]:9999`

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
Fixes #120.

(Classified as a bug, but unimportant so included in a minor release.)

### CODE REVIEW

File: src/swarm/neo/protocol/connect/ConnectProtocol.d [hunk 1]
```diff
@@ -519,8 +519,13 @@ class ConnectProtocol: ISelectClient
 
     override public void finalize ( FinalizeStatus status )
     {
+        static Exception finalize_exception;
+        if ( finalize_exception is null )
+            finalize_exception = new Exception("ConnectProtocol: finalized",
```
Comment (nemanja-boric-sociomantic) [inline src/swarm/neo/protocol/connect/ConnectProtocol.d:524]: ouch

**Review by nemanja-boric-sociomantic — APPROVED** [review]

File: src/swarm/neo/protocol/connect/ConnectProtocol.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/swarm/neo/protocol/connect/ConnectProtocol.d:524]: Ouch?

Comment (gavin-norman-sociomantic) [reply src/swarm/neo/protocol/connect/ConnectProtocol.d:524]: Oh, ouch before the fix ;)

Comment (nemanja-boric-sociomantic) [reply src/swarm/neo/protocol/connect/ConnectProtocol.d:524]: Referring the the previous state of affairs.

**Pull request merged**